services/convert/convert.go

```diff
@@ -105,33 +105,60 @@ func ToBranch(ctx context.Context, repo *repo_model.Repository, branchName strin
 	return branch, nil
 }
 
-// ToBranchProtection convert a ProtectedBranch to api.BranchProtection
-func ToBranchProtection(ctx context.Context, bp *git_model.ProtectedBranch) *api.BranchProtection {
-	pushWhitelistUsernames, err := user_model.GetUserNamesByIDs(ctx, bp.WhitelistUserIDs)
-	if err != nil {
-		log.Error("GetUserNamesByIDs (WhitelistUserIDs): %v", err)
-	}
-	mergeWhitelistUsernames, err := user_model.GetUserNamesByIDs(ctx, bp.MergeWhitelistUserIDs)
-	if err != nil {
-		log.Error("GetUserNamesByIDs (MergeWhitelistUserIDs): %v", err)
+// getWhitelistEntities returns the names of the entities that are in the whitelist
+func getWhitelistEntities(entities []any, whitelistIDs []int64) []string {
```

Comment:
Could it use generic to avoid misuse?

```
func getWhitelistEntities[T *user_model.User|*organization.Team](entities []T ...)
```

Reply:
Generic was my first idea. Unfortunately, generics don't support `T.ID` and `T.Name`.

Reply:
```
type S1 struct {
	ID int
}
type S2 struct {
	Name string
}

func F[T *S1|*S2] (v T) {
	switch v :=  any(v).(type) {
	case *S1:
		println(v.ID)
	case *S2:
		println(v.Name)
	}
}
```